### PR TITLE
Finished up Edit/Retry, including showing prompt for Document Code

### PR DESCRIPTION
**Pull request by steveyegge**
Edit/Retry in JetBrains needed to show the "instruction" (prompt) generated for Document Code; i.e., "Write a brief documentation comment for the selected code. If documentation comments exist..." etc.

I used a pretty quick-and-dirty solution. It just goes along with the EditTask as an optional string for the client to consume, and we send it on every editTask/didUpdate notification. That way it's available if the user chooses to Edit/Retry.

Most of the work for this is in a sourcegraph/jetbrains PR coming up shortly.

## Test plan

The test plan is in TESTING.md.

### CODE REVIEW

File: agent/src/AgentFixupControls.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export class AgentFixupControls implements FixupControlApplicator {
             state: task.state,
             error: errorToCodyError(task.error),
             selectionRange: task.selectionRange,
+            instruction: task.instruction.toString().trim()
```
Comment (abeatrix) [inline agent/src/AgentFixupControls.ts:54]:
```suggestion
            instruction: task.instruction?.trim()
```

✋ q: do we need to add ? to check if the instruction string is undefined before we trim it? 

(Ignore me if there was a check somewhere that I missed 🙈)

**Review by abeatrix — APPROVED** [review]

File: agent/src/AgentFixupControls.ts (hunk 1 above)
Comment (pkukielka) [reply agent/src/AgentFixupControls.ts:54]: Why do we need to trim it at all here, shouldn't we do that when it is put into `FixupTask`?
Anyway LGTM

**Review by pkukielka — APPROVED** [review]
LGTM

File: agent/src/AgentFixupControls.ts (hunk 1 above)
Comment (steveyegge) [reply agent/src/AgentFixupControls.ts:54]: good catch; I'll add that.

**Pull request merged**